agent/internal/client/base_client.go

```diff
@@ -492,6 +492,7 @@ func (c *baseCommunicator) GetCedarConfig(ctx context.Context) (*apimodels.Cedar
 	if err != nil {
 		return nil, util.RespErrorf(resp, errors.Wrap(err, "getting the Cedar config").Error())
 	}
+	defer resp.Body.Close()
```

Comment:
This isn't necessary because `utility.ReadJSON` closes the body already (same with the others in this function). Since most of the methods share the same logic (make request, read the JSON body) is there a reason some of the methods have been changed to defer close the body and others have been changed to nolint?

Reply:
It looks like the `bodyclose` linter dislikes our`utility.ReadJSON` calls closing the body and triggers a false positive. I tried finding solutions but it seems like we either have to double close it (which I think is sloppy) or apply //nolint _everywhere_ so it doesn't throw. I think (at least for now), we shouldn't add this linter. I'll adjust the PR to remove the double closes and not add the linter.